# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: third_party/rust/audioipc-client/src/context.rs
# Commit: 634412ce2943
# Full Hash: 634412ce29430557eec6d0c0313f1908e61ad13d
# Author: Matthew Gregan <kinetik@flim.org>
# Date: 2021-08-11 03:46:10
# Regressor Bug: 1724141
# File Overlap Count: 6
# Description:
#   Bug 1724141 - Update audioipc to 8bb1a227.  r=chunmin
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D121826
# ==============================================================================

diff -r 722284bed619 -r 634412ce2943 third_party/rust/audioipc-client/src/context.rs
--- a/third_party/rust/audioipc-client/src/context.rs	Tue Aug 10 20:38:51 2021 +0000
+++ b/third_party/rust/audioipc-client/src/context.rs	Tue Aug 10 20:52:03 2021 +0000
@@ -11,8 +11,7 @@
 #[cfg(not(target_os = "linux"))]
 use audio_thread_priority::promote_current_thread_to_real_time;
 use audioipc::codec::LengthDelimitedCodec;
-use audioipc::frame::{framed, Framed};
-use audioipc::platformhandle_passing::{framed_with_platformhandles, FramedWithPlatformHandles};
+use audioipc::framing::{framed, Framed};
 use audioipc::{core, rpc};
 use audioipc::{
     messages, messages::DeviceCollectionReq, messages::DeviceCollectionResp, ClientMessage,
@@ -38,10 +37,8 @@
 impl rpc::Client for CubebClient {
     type Request = ServerMessage;
     type Response = ClientMessage;
-    type Transport = FramedWithPlatformHandles<
-        audioipc::AsyncMessageStream,
-        LengthDelimitedCodec<Self::Request, Self::Response>,
-    >;
+    type Transport =
+        Framed<audioipc::AsyncMessageStream, LengthDelimitedCodec<Self::Request, Self::Response>>;
 }
 
 pub const CLIENT_OPS: Ops = capi_new!(ClientContext, ClientStream);
@@ -190,7 +187,7 @@
             stream: audioipc::AsyncMessageStream,
             tx_rpc: &mpsc::Sender<rpc::ClientProxy<ServerMessage, ClientMessage>>,
         ) {
-            let transport = framed_with_platformhandles(stream, Default::default());
+            let transport = framed(stream, Default::default());
             let rpc = rpc::bind_client::<CubebClient>(transport);
             // If send fails then the rx end has closed
             // which is unlikely here.
@@ -206,7 +203,7 @@
         let thread_destroy_callback = params.thread_destroy_callback;
 
         let server_stream =
-            unsafe { audioipc::MessageStream::from_raw_fd(params.server_connection) };
+            unsafe { audioipc::MessageStream::from_raw_handle(params.server_connection) };
 
         let core = core::spawn_thread(
             "AudioIPC Client RPC",
@@ -368,15 +365,15 @@
         assert_not_in_callback();
 
         if !self.device_collection_rpc {
-            let fds = send_recv!(self.rpc(),
+            let mut fd = send_recv!(self.rpc(),
                                  ContextSetupDeviceCollectionCallback =>
                                  ContextSetupDeviceCollectionCallback())?;
 
-            // TODO: The lowest comms layer expects exactly 3 PlatformHandles, but we only
-            // need one here.  The server sent two dummy valid handles, ignore those (closed on drop)
-            // and use the one we need.
-            let stream =
-                unsafe { audioipc::MessageStream::from_raw_fd(fds.platform_handles[0].into_raw()) };
+            let stream = unsafe {
+                audioipc::MessageStream::from_raw_handle(
+                    fd.platform_handle.take_handle().into_raw(),
+                )
+            };
 
             let server = DeviceCollectionServer {
                 input_device_callback: self.input_device_callback.clone(),